src/api-response.ts

```diff
@@ -1,17 +1,34 @@
 import { z } from "zod";
-import { MimeDefinition, mimeJson } from "./mime";
 
-export type ApiResponse<A = z.ZodTypeAny> = {
-  schema: A;
-  mimeTypes: string[];
-};
+export type ApiResponse<S extends z.ZodType> = {
+  schema: S;
+  /**
+   * @default 200 for a positive response
+   * @default 400 for a negative response
+   * */
+  statusCode?: number;
+} & (
+  | {
+      /** @default [ "application/json" ] */
+      mimeTypes?: [string, ...string[]];
+    }
+  | {
+      /** @default "application/json" */
+      mimeType?: string;
```

Comment:
this does not actually work.
typescript fails to perform this check on the function result